src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,412 @@
+        )
+
+    (target_var,) = y.data_vars
+    independent_var = list(x.data_vars)
+
+    if y_pred is None:
+        y_pred = target_var
+
+    if isinstance(y_pred, xr.Dataset):
+        raise TypeError(
+            "y_pred can't be a dataset because multiple target variables are not supported yet."
+        )
+
+    if not isinstance(y_pred, xr.DataArray):
+        y_pred = process_group_variables_coords(
+            dt,
+            group=group,
+            var_names=y_pred,
+            filter_vars=filter_vars,
+            coords=coords,
+        )
+
+    if x_pred is None:
+        x_pred = independent_var
+
+    if not isinstance(x_pred, xr.DataArray | xr.Dataset):
+        if group == "predictions":
+            x_pred = process_group_variables_coords(
+                dt,
+                group="predictions_constant_data",
+                var_names=x_pred,
+                filter_vars=filter_vars,
+                coords=coords,
+            )
+        else:
+            x_pred = x
+
+    if isinstance(ci_prob, Sequence):
+        x_with_prob = x.expand_dims(dim={"prob": ci_prob})
+    else:
+        x_with_prob = x
+
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+    bg_color = plot_bknd.get_background_color()
+    _, contrast_gray_color = get_contrast_colors(bg_color=bg_color, gray_flag=True)
+    if plot_collection is None:
+        pc_kwargs.setdefault("cols", "__variable__")
+        pc_kwargs["figure_kwargs"] = pc_kwargs.get("figure_kwargs", {}).copy()
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        if isinstance(ci_prob, Sequence):
+            alpha_dims = pc_kwargs["aes"].get("alpha", None)
+            if alpha_dims is None:
+                pc_kwargs["aes"].setdefault("alpha", ["prob"])
+                pc_kwargs["alpha"] = np.linspace(0.1, 0.5, len(ci_prob))
+            else:
+                warnings.warn(
+                    "When multiple credible intervals are plotted, "
+                    "it is recommended to map 'alpha' aesthetic to 'prob' "
+                    "dimension to differentiate between intervals.",
+                )
+
+        pc_kwargs = set_wrap_layout(pc_kwargs, plot_bknd, x)
+        plot_collection = PlotCollection.wrap(
+            x_with_prob,
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    if aes_by_visuals is None:
+        aes_by_visuals = {}
+    else:
+        aes_by_visuals = aes_by_visuals.copy()
+    aes_by_visuals.setdefault("line", plot_collection.aes_set.difference("alpha"))
+    if isinstance(ci_prob, Sequence):
+        aes_by_visuals.setdefault("ci_line", {"alpha"})
+        aes_by_visuals.setdefault("ci_fill", {"alpha"})
+
+    # calculations for credible interval
+    ci_fun = azs.hdi if ci_kind == "hdi" else azs.eti
+    ci_dims, _, _ = filter_aes(plot_collection, aes_by_visuals, "ci_fill", sample_dims)
```

Comment:
you can already define `fill_aes` and `fill_ignore` here and skip calling it inside the `if fill_kwargs is not False`